libs/datasets/timeseries.py

```diff
@@ -877,10 +924,10 @@ def drop_regions_without_population(
     known_location_id_to_drop: Sequence[str],
     log: Union[structlog.BoundLoggerBase, structlog._config.BoundLoggerLazyProxy],
 ) -> MultiRegionDataset:
-    # latest_population is a Series with location_id index
-    latest_population = mrts._latest_data[CommonFields.POPULATION]
-    locations_with_population = mrts._latest_data.loc[latest_population.notna()].index
-    locations_without_population = mrts._latest_data.loc[latest_population.isna()].index
+    assert mrts._regional_attributes.index.names == [CommonFields.LOCATION_ID]
```

Comment:
Should this be part of the public api? don't love accessing the "private" property outside of the class

Reply:
> This is a good next step!
> 
> One thing I'd like to work on is figuring out the best ways to remove the necessity for the various geodata columns and adding accessors onto multiregion dataset and one region dataset. for example, places where we group by state, can we have a method `groupby_parent_region` that walks up the location_id one level.

> 
> another thing I would love to think about how to do is push more of the data to the region itself. For example: `region.county` and `region.state` seem more well placed than having to pull from a dictionary of that data. I think that would help reduce some of the logic around that shuffling


Yes, I'm trying to add accessors onto multiregion dataset and one region dataset to reduce direct use of the dataframes. One big benefit of having all the data in DataFrames (at least internal MultiRegionDataset) is it provides an easy fast way to copy to and from disk.

Reply:
Accessing private / single underscore attributes outside the class is a bit ugly, but I'm trying to discourage adding new dependencies on them... while adding more myself.